eventsource/repository.go

```diff
@@ -104,10 +110,17 @@ func (repo *repository) SaveWithContext(ctx context.Context, events ...Event) (e
 		})
 	}
 
-	if err = repo.store.SaveWithContext(ctx, records...); err != nil {
+	tx, err := repo.store.NewTransaction(ctx, records...)
+	if err != nil {
+		return
+	}
+
+	if err = tx.Save(); err != nil {
 		return
 	}
 
+	rollback = tx.Rollback
+
 	for idx, record := range records {
```

Comment:
it looks like `OnSave` will be called even if a rollback is executed, would that be a problem?

Reply:
Yeah, that could be a problem. I solved it, no need to keep it. Most of the time you want to pass other arguments to OnSave, therefore you can do this outside the package, before calling Save or after.